wdl-engine/src/inputs.rs

```diff
@@ -311,7 +312,15 @@ impl Serialize for TaskInputs {
         S: serde::Serializer,
     {
         // Only serialize the input values
-        self.inputs.serialize(serializer)
+        let mut map = serializer.serialize_map(Some(self.inputs.len()))?;
+        self.inputs.iter().for_each(|(k, v)| {
+            let serialized_value = v.serializable_with_pairs();
+            map.serialize_entry(k, &serialized_value)
+                .unwrap_or_else(|e| {
+                    panic!("failed to serialize value `{v}` for input `{k}`: {e}");
```

Comment:
I'd be more comfortable bubbling up the serialization error here instead of panicking.